### PR TITLE
Fix #2238: Test added for queue name verification

### LINKED ISSUE #2238: Update method to store queue name in Challenge model
Currently, we store the queue name for each challenge using something like this:

```
challenge_queue_name = "{}-{}-{}".format(challenge_title, challenge_pk, random_challenge_id)[:75]
```

Where the challenge title can be any string. But SQS does not allow special characters in the queue name and hence we should validate the queue name before storing it. 

## Deliverables

- [ ] Write a utils function that generates a valid SQS queue name given a string as an input
- [ ] Use the function written in the previous step to generate the queue name for a challenge
- [ ] Write tests to check if the function created in the first step is robust enough

### PR DESCRIPTION
**Pull request by pushkalkatara**
Added tests for issue #2238 

Test Results:

```
.[2019-09-24 07:38:51] INFO evaluation_script for 487 is added first time !
.[2019-09-24 07:38:51] INFO evaluation_script for 488 is added first time !
.[2019-09-24 07:38:51] INFO evaluation_script for 489 is added first time !
............E.......E.....
======================================================================
ERROR: test_process_submission_message_succesfully (tests.integration.worker.test_submission_worker.ProcessSubmissionCallbackTestClass)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/mock/mock.py", line 1305, in patched
    return func(*args, **keywargs)
  File "/code/tests/integration/worker/test_submission_worker.py", line 173, in test_process_submission_message_succesfully
    submission_worker.process_submission_message(message)
  File "/code/scripts/workers/submission_worker.py", line 563, in process_submission_message
    submission_instance = extract_submission_data(submission_id)
  File "/code/scripts/workers/submission_worker.py", line 307, in extract_submission_data
    submission_input_file, submission_input_file_path
  File "/code/scripts/workers/submission_worker.py", line 127, in download_and_extract_file
    with open(download_location, "wb") as f:
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmp5lv6cp8c/compute/submission_files/submission_6/user_annotation_file.txt'

======================================================================
ERROR: test_error_handlers (tests.unit.web.test_views.TestErrorPages)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/code/tests/unit/web/test_views.py", line 174, in test_error_handlers
    response = page_not_found(request)
  File "/code/apps/web/views.py", line 33, in page_not_found
    response = render(request, "error404.html")
  File "/usr/local/lib/python3.6/site-packages/django/shortcuts.py", line 30, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "/usr/local/lib/python3.6/site-packages/django/template/loader.py", line 68, in render_to_string
    return template.render(context, request)
  File "/usr/local/lib/python3.6/site-packages/django/template/backends/django.py", line 66, in render
    return self.template.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 205, in render
    with context.bind_template(self):
  File "/usr/local/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/usr/local/lib/python3.6/site-packages/debug_toolbar/panels/templates/panel.py", line 49, in _request_context_bind_template
    context = processor(self.request)
  File "/usr/local/lib/python3.6/site-packages/django/template/context_processors.py", line 43, in debug
    if settings.DEBUG and request.META.get('REMOTE_ADDR') in settings.INTERNAL_IPS:
AttributeError: 'HttpResponseNotFound' object has no attribute 'META'

======================================================================
ERROR: middleware.metrics (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: middleware.metrics
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/unittest/loader.py", line 462, in _find_test_path
    package = self._get_module_from_name(name)
  File "/usr/local/lib/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/code/middleware/metrics/__init__.py", line 1, in <module>
    from .metrics_middleware import DatadogMiddleware
  File "/code/middleware/metrics/metrics_middleware.py", line 7, in <module>
    from datadog import initialize
ModuleNotFoundError: No module named 'datadog'


======================================================================
FAIL: test_extract_submission_data_succesfully (tests.integration.worker.test_submission_worker.ExtractSubmissionDataTestClass)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/mock/mock.py", line 1305, in patched
    return func(*args, **keywargs)
  File "/code/tests/integration/worker/test_submission_worker.py", line 233, in test_extract_submission_data_succesfully
    mock_down_ext.assert_called_with(submission_input_file_url, submission_input_file_path)
  File "/usr/local/lib/python3.6/site-packages/mock/mock.py", line 937, in assert_called_with
    six.raise_from(AssertionError(_error_message(cause)), cause)
  File "<string>", line 3, in raise_from
AssertionError: Expected call: download_and_extract_file('http://testserver/media/submission_files/submission_2/af1baa77-b6a0-4585-a3af-133a9f935bc9.txt', 'mocked/dir/submission_2/af1baa77-b6a0-4585-a3af-133a9f935bc9.txt')
Actual call: download_and_extract_file('http://django:8000/media/submission_files/submission_2/af1baa77-b6a0-4585-a3af-133a9f935bc9.txt', 'mocked/dir/submission_2/af1baa77-b6a0-4585-a3af-133a9f935bc9.txt')

----------------------------------------------------------------------
Ran 434 tests in 57.040s

FAILED (failures=1, errors=3)
Destroying test database for alias 'default'...
```

### CODE REVIEW

**Comment by pushkalkatara** [comment]
@RishabhJain2018 any idea why build is failing?

**Comment by Sanji515** [comment]
> @RishabhJain2018 any idea why build is failing?

@pushkalkatara You got one flake8 error here https://travis-ci.org/Cloud-CV/EvalAI/builds/588825684#L1264, please resolve it in order to pass the build.

**Comment by pushkalkatara** [comment]
Oh silly. Thanks for pointing out @Sanji515 . Resolved it.

**Comment by RishabhJain2018** [comment]
@KhalidRmb Can you please review this?

File: tests/unit/challenges/test_utils.py [hunk 1]
```diff
@@ -1,18 +1,48 @@
...
+        title = ''.join([random.choice(string.printable) for i in range(256)])
+        sqs_queue_name = get_queue_name(title)
+        self.assertNotRegex(sqs_queue_name, "[^a-zA-Z0-9_-]")
+        self.assertLessEqual(len(sqs_queue_name), 80)
+
+    def test_sqs_queue_name_generator_empty_title(self):
+        title = ""
+        sqs_queue_name = get_queue_name(title)
+        self.assertNotRegex(sqs_queue_name, "[^a-zA-Z0-9_-]")
+        self.assertLessEqual(len(sqs_queue_name), 80)
```
Comment (Ram81) [inline tests/unit/challenges/test_utils.py]: @pushkalkatara shouldn't we have a check for `len(sqs_queue_name)>0` ?

Comment (pushkalkatara) [reply tests/unit/challenges/test_utils.py]: Yes that's a good idea. Would you like to make the change?

**Pull request merged**